synapse/groups/groups_server.py

```diff
@@ -67,6 +67,10 @@ def check_group_is_ours(self, group_id, and_exists=False, and_is_admin=None):
         if and_exists and not group:
             raise SynapseError(404, "Unknown group")
 
+        is_user_in_group = yield self.store.is_user_in_group(requester_user_id, group_id)
+        if is_user_in_group or not group.is_public:
+            raise SynapseError(404, "Unknown group")
```

Comment:
We should probably only do this if the group exists, in particular this currently doesn't work well with create_group. (Speaking of which, you forgot to add requester_user_id to create_group, which is why its failing)

Reply:
Shouldn't that also be `if not is_user_in_group...`?

Reply:
yes o\